src/main/java/hudson/remoting/ChannelBuilder.java

```diff
@@ -188,11 +190,22 @@ public boolean isRemoteClassLoadingAllowed() {
         return remoteClassLoadingAllowed;
     }
 
-    public ChannelBuilder withJarCache(JarCache jarCache) {
+    /**
+     * Sets the JAR cache storage.
+     * @param jarCache JAR Cache to be used. If {@code null}, a default value will be used by the {@link Channel}
+     * @return {@code this}
+     */
+    public ChannelBuilder withJarCache(@CheckForNull JarCache jarCache) {
         this.jarCache = jarCache;
         return this;
     }
 
+    /**
+     * Gets the JAR Cache storage.
+     * @return {@code null} if it is not defined.
+     *         {@link Channel} implementation should use a default cache value then.
+     */
+    @CheckForNull
```

Comment:
Undocumented risk of null in the previous API. In open-source I see external usages only in ATH and Remoting-based CLI. Better to have it annotated anyway.